ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+# Orleans and Akka Actors: A Comparison
+
+## Step-By-Step Review
+
+This overview walks along the Technical Report [MSR-TR-2014-41](http://research.microsoft.com/apps/pubs/default.aspx?id=210931) by *Philip A. Bernstein, Sergey Bykov, Allan Geller, Cabriel Kliot, Jorgen Thelin (Microsoft Research)*, for another comparison focusing on Erlang and Riak see [Christopher Meiklejohn’s excellent article](http://christophermeiklejohn.com/orleans.html).
+
+### Introduction
+
+Different focus:
+
+  * Akka as a toolkit for building distributed systems, offering the full power but also exposing the inherent essential complexity.
+  
+  * Orleans restricts applicability in order to allow seamless use without understanding distributed computing.
+  
+#### Lifecycle
+
+  * Orleans Grains do not have a lifecycle, can neither be started nor stopped. Consequently they also cannot fail and be restarted, therefore Orleans does not offer tools for software failure handling—the failure handling aspects concentrate on recovering from hardware crashes.
+  
+  * Akka Actors implement the full model including defined lifecycle beginning and end, these are explicit operations. Persistent Actors support extending the lifecycle of a logical unit of computation beyond the lifecycle of the running process instance. Restarting an Actor provides powerful means for automatic service recovery.
+  
+#### Automatic Creation
+
+  * Orleans Grains are created automatically whenever needed, which implies that initialization activities cannot have persistent effects—otherwise these would be uncontrollable. OTOH this frees the user from having to consider the need for Grain creation. It also means that asking for a Grain you don’t know which implementation you will get.
```

Comment:
Ah, understood. What I meant was that “fire all the missiles” (or “charge the credit card”) are not sensible things to do within OnActivateAsync since triggering these external actions presumably should be driven by explicitly intent and since the activation lifecycle cannot be explicitly controlled this would be unreliable. The reason why I said this (which I should also include in the text) is that Actors can be used as process managers that perform external activities explicitly and in a controlled fashion based on lifecycle events, and since I want to highlight differences between the two approaches I’d like to mention this (if true).